tests/test_javascript.py

```diff
@@ -83,3 +83,25 @@ def test_beware_infinite_loop(lexer):
         (Token.Text, '\n'),
     ]
     assert list(lexer.get_tokens(fragment)) == tokens
+
+@pytest.fixture(scope='module')
+def ts_lexer():
+    yield TypeScriptLexer()
+
+def test_function_definition(ts_lexer):
+    fragment = u'async function main() {\n}'
+    tokens = [
+        (Token.Name.Other, u'async'),
```

Comment:
```suggestion
        (Token.Keyword, u'async'),
```
The `async` should be a keyword (https://github.com/pygments/pygments/pull/1651)